File: Java-Core/src/main/java/presto/python/PythonArgumentList.java
package presto.python;

import presto.utils.CodeWriter;
import presto.utils.ObjectList;

public class PythonArgumentList extends ObjectList<PythonArgument> {
	
	private static final long serialVersionUID = 1L;

	public PythonArgumentList() {
	}

	public PythonArgumentList(PythonArgument arg) {
		this.add(arg);
	}

	public void toDialect(CodeWriter writer) {
		if(this.size()>0) {
			for(PythonArgument arg : this) {
				arg.toDialect(writer);
				writer.append(", ");
			}
			writer.trimLast(2);
		}
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/python/PythonArgumentList.js
var ObjectList = require("../utils/ObjectList").ObjectList;

function PythonArgumentList(argument) {
    ObjectList.call(this);
    argument = argument || null;
    if(argument!==null) {
        this.add(argument);
    }
    return this;
}

PythonArgumentList.prototype = Object.create(ObjectList.prototype);
PythonArgumentList.prototype.constructor = PythonArgumentList;

PythonArgumentList.prototype.toDialect = function(writer) {
    if(this.length>0) {
        for(var i=0;i<this.length;i++) {
            this[i].toDialect(writer);
            writer.append(", ");
        }
        writer.trimLast(2);
    }
};

exports.PythonArgumentList = PythonArgumentList;